apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+import { Spinner } from "@/components/ui/Spinner/Spinner";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { CheckIcon, ChevronsUpDown } from "lucide-react";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useMemo } from "react";
+import invariant from "tiny-invariant";
+import { useEcosystemList } from "../../../hooks/use-ecosystem-list";
+
+export function EcosystemHeader({ ecosystemId }: { ecosystemId: string }) {
+  const { ecosystems, isFetched } = useEcosystemList();
+  const router = useRouter();
+  const currentEcosystem = useMemo(
+    () => ecosystems?.find((ecosystem) => ecosystem.id === ecosystemId),
+    [ecosystems, ecosystemId],
+  );
+
+  if (
+    isFetched &&
+    !ecosystems?.find((ecosystem) => ecosystem.id === ecosystemId)
+  ) {
+    // No ecosystem found, redirect to 404
+    router.replace("/404");
+  }
+
+  if (!currentEcosystem) {
+    return (
+      <div className="container flex flex-col gap-8 px-4 py-6">
+        <Spinner className="size-8" />
+      </div>
+    );
+  }
+
+  invariant(currentEcosystem, "Ecosystem not found");
```

Comment:
why is this needed? there's already the if statement above which should achieve the same thing?

Reply:
It's just to make make typescript happy

Reply:
should already be happy because of the if above?

Reply:
It appears I was imagining things